rust/otel-arrow-rust/src/pdata/mod.rs

```diff
@@ -10,12 +10,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! This module contains data structures for OTLP and OTAP pipeline data
+//! This module contains data structures for OTLP and OTAP pipeline data.
+///
+/// - Visitors for primitive data types (string, i64, ...)
+/// - SliceVisitor<_, _>
+/// - NoopVisitor
+/// - From<_> for TraceID, SpanID
+pub mod otlp;
```

Comment:
It looks like these comments are related to this `pdata` module, not the `pdata::otlp` child module, right? If so, I think we might want these comments to be prefixed by `//!` instead of `///`. 

```suggestion
//! This module contains data structures for OTLP and OTAP pipeline data.
//!
//! - Visitors for primitive data types (string, i64, ...)
//! - SliceVisitor<_, _>
//! - NoopVisitor
//! - From<_> for TraceID, SpanID

pub mod otlp;
```

My understanding is that `//!` is inner doc comment, documenting what the comment is currently inside of (in this case, the `pdata` module), vs `///` is the outer doc comment, documenting what the comment proceeds